### ISSUE TITLE
defunct processes not cleaned up by rackspace-monitoring-agent

### ISSUE THREAD

**Issue by boxidau** [virgo-agent-toolkit/rackspace-monitoring-agent#789]
I've just been diagnosing some chef related error on a server it was caused by a huge amount of defunct processes on CentOS 6.6

rackspace-monitoring-agent is configured to use the following plugin https://github.com/racker/rackspace-monitoring-agent-plugins-contrib/blob/master/chef_node_checkin.py

via the following server side config file

```
$~ cat /etc/rackspace-monitoring-agent.conf.d/monitoring-plugin-chef-client.yaml 
# CHEF MANAGED FILE: DO NOT EDIT
# Controlling Cookbook: platformstack
#

type: agent.plugin
label: chef-client
disabled: false
period: 60
timeout: 30
details:
    file: chef_node_checkin.py
    args: []
    timeout: 60
alarms:
    alarm-queue:
        label: 
        notification_plan_id: npMANAGED
        criteria: |        
```

an excerpt from `ps ax`:

```
32172 ?        Z      0:00 [chef_node_check] <defunct>
32186 ?        Z      0:00 [chef_node_check] <defunct>
32189 ?        Z      0:00 [chef_node_check] <defunct>
32190 ?        Z      0:00 [chef_node_check] <defunct>
32203 ?        Z      0:00 [chef_node_check] <defunct>
32204 ?        Z      0:00 [chef_node_check] <defunct>
32212 ?        Z      0:00 [chef_node_check] <defunct>
32220 ?        Z      0:00 [chef_node_check] <defunct>
32224 ?        Z      0:00 [chef_node_check] <defunct>
32226 ?        Z      0:00 [chef_node_check] <defunct>
```

`service rackspace-monitoring-agent restart` clears the process list

Any thoughts?

**Comment by rphillips** [comment]
Thank you for the bug report. I have a fix [1], and it will be released in the next version of the agent.

[1] https://github.com/luvit/luvit/pull/780

**Comment by rphillips** [comment]
If you could try out the package here [1] that would be a great help.

1. http://unstable.packages.cloudmonitoring.rackspace.com/

**Comment by boxidau** [comment]
Thanks @rphillips I'll try it out :+1:

**Comment by boxidau** [comment]
Fixes this issue, thanks :) Any idea when the next release is scheduled for?

**Comment by rphillips** [comment]
Just released it now as 2.1.5.
